File: solutions/49GroupAnagrams/main.go
/*
Given an array of strings strs, group the anagrams together.
You can return the answer in any order.
*/

func groupAnagrams(strs []string) [][]string {
	groups := make(map[string][]string)
	for _, str := range strs {
		runes := []rune(str)
		sort.Slice(runes, func(i, j int) bool { return runes[i] < runes[j] })
		key := string(runes)
		groups[key] = append(groups[key], str)
	}
	result := [][]string{}
	for _, group := range groups {
		result = append(result, group)
	}
	return result
}

File: solutions/49GroupAnagrams/main.py
'''
Given an array of strings strs, group the anagrams together.
You can return the answer in any order.
'''

class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        groups = {}
        for str in strs:
            key = "".join(sorted(str))
            if key not in groups:
                groups[key] = []
            groups[key].append(str)
        result = []
        for group in groups.values():
            result.append(group)
        return result